### DISCUSSION TITLE
How to customize text in the right click menu of TextBox

### DISCUSSION THREAD

**Discussion by msnoshain** [AvaloniaUI/Avalonia#9832] [Q&A] [answered]
I want to customize text in the right click menu of TextBox

![image](https://user-images.githubusercontent.com/85274323/210178753-dd55b5e4-6ae6-4eb8-af3a-e001bf20e301.png)

So I write a style like: 

`<Style Selector="TextBox /template/ MenuItem#TextBoxContextFlyoutCutItem">`
`    <Setter Property="Header" Value="MyCut"/>`
`</Style>`

I also tried:

`<Style Selector="TextBox /template/ MenuItem#TextBoxContextMenuCutItem">`
`    <Setter Property="Header" Value="MyCut"/>`
`</Style>`

None of them worked

**Comment by timunie** [comment, marked as answer]
I think you need to override this: https://github.com/AvaloniaUI/Avalonia/blob/401bc521d2947bf7699a530f3d6e43e7a32c6baf/src/Avalonia.Themes.Default/TextBox.xaml#L7-L11

and https://github.com/AvaloniaUI/Avalonia/blob/401bc521d2947bf7699a530f3d6e43e7a32c6baf/src/Avalonia.Themes.Default/TextBox.xaml#L27

**Comment by msnoshain** [comment]
Thanks. I declare a MenuFlyout Resource, and set the value of ContextFlyout Property. It works.

**Reply by DamianSuess** [reply]
@msnoshain out of curiosity, could you provide a snippet of how you implemented this?

**Comment by msnoshain** [comment]
> @msnoshain out of curiosity, could you provide a snippet of how you implemented this?

sure, see [here](https://github.com/msnoshain/ChatAI/blob/master/AvaloniaApplication1/AvaloniaApplication1/Views/MainView.axaml)

**Reply by DamianSuess** [reply]
@msnoshain Ah, now I see what I was doing wrong. Thank you!
